include/zephyr/cache.h

```diff
@@ -441,6 +441,97 @@ static ALWAYS_INLINE size_t sys_cache_instr_line_size_get(void)
 #endif
 }
 
+/**
+ * @brief Test if a pointer is in cached region.
+ *
+ * Some hardware may map the same physical memory twice
+ * so that it can be seen in both (incoherent) cached mappings
+ * and a coherent "shared" area. This tests if a particular
+ * pointer is within the cached, coherent area.
+ *
+ * @param ptr Pointer
+ *
+ * @retval True if pointer is in cached region.
+ * @retval False if pointer is not in cached region.
+ */
+static ALWAYS_INLINE bool sys_cache_is_ptr_cached(void *ptr)
+{
+#if defined(CONFIG_CACHE_MANAGEMENT) && defined(CONFIG_CACHE_COHERENCY)
+	return cache_is_ptr_cached(ptr);
+#endif
+	ARG_UNUSED(ptr);
+
+	return false;
+}
+
+/**
+ * @brief Test if a pointer is in un-cached region.
+ *
+ * Some hardware may map the same physical memory twice
+ * so that it can be seen in both (incoherent) cached mappings
+ * and a coherent "shared" area. This tests if a particular
+ * pointer is within the un-cached, incoherent area.
+ *
+ * @param ptr Pointer
+ *
+ * @retval True if pointer is not in cached region.
+ * @retval False if pointer is in cached region.
+ */
+static ALWAYS_INLINE bool sys_cache_is_ptr_uncached(void *ptr)
+{
+#if defined(CONFIG_CACHE_MANAGEMENT) && defined(CONFIG_CACHE_COHERENCY)
+	return cache_is_ptr_uncached(ptr);
+#endif
+	ARG_UNUSED(ptr);
+
+	return false;
+}
+
+/**
+ * @brief Return cached pointer to a RAM address
+ *
+ * This function takes a pointer to any addressable object (either in
+ * cacheable memory or not) and returns a pointer that can be used to
+ * refer to the same memory through the L1 data cache.  Data read
```

Comment:
Nitpick, but this documentation is a lie with the new scheme: it returns the same pointer if !CACHE_COHERENCY, which may or may not be cached depending on the hardware/OS setup, but certainly no promises are being made that access goes through an L1 data cache.